frontend/views/containers/contributions/SupportHistory.vue

```diff
@@ -41,22 +42,24 @@ export default ({
       'currentPaymentPeriod',
       'withGroupCurrency',
       'groupTotalPledgeAmount',
-      'groupCreatedDate'
+      'groupCreatedDate',
+      'thisPeriodPaymentInfo'
     ])
   },
   mounted () {
     this.updateHistory()
   },
   methods: {
     async updateHistory () {
-      const periods = await this.getAllSortedPeriodKeys()
-      this.history = await Promise.all(periods.map(async (period, i) => {
+      const allPeriods = await this.getAllSortedPeriodKeys()
+      const periods = allPeriods.slice(-MAX_HISTORY_PERIODS)
+      this.history = await Promise.all(periods.map(async (period) => {
         const totalTodo = await this.getTotalTodoAmountForPeriod(period)
         const totalDone = await this.getTotalPledgesDoneForPeriod(period)
 
         return {
-          total: totalDone === 0 ? 0 : totalDone / totalTodo,
-          title: this.getPeriodFromStartToDueDate(period),
+          total: totalDone === 0 || totalTodo === 0 ? 0 : totalDone / totalTodo,
```

Comment:
Nice one here.(I remember fixing `NaN` being displayed somewhere else by adding this there)